src/app/EthersContext.jsx

```diff
@@ -56,6 +63,22 @@ export function EthersProvider({ children }) {
 					setError(e.message);
 					setStatus(EthersStatus.WRONG_NETWORK);
 				}
+			} else {
```

Comment:
I feel this could be simplified. You only need the if/else to decide the provider. After that the contract instances can be constructed the same. I know, in the metamask version I'm passing a signer which you don't have, but in fact the ethers.js Contract constructor accepts both signer and provider as the third argument. 
Simple ugly solution would be to pass the provider in the signer option. Alternatively and more cleanly, you could update the getContract method to accept a provider option.